src/Logging/Logging.Console/src/ConsoleLogger.cs

```diff
@@ -109,23 +134,77 @@ public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId,
             }
 
             var timestampFormat = Options.TimestampFormat;
-            // Queue log message
-            _queueProcessor.EnqueueMessage(new LogMessageEntry(
+
+            return new LogMessageEntry(
+                message: logBuilder.ToString(),
                 timeStamp: timestampFormat != null ? DateTime.Now.ToString(timestampFormat) : null,
                 levelString: logLevelString,
                 levelBackground: logLevelColors.Background,
                 levelForeground: logLevelColors.Foreground,
                 messageColor: DefaultConsoleColor,
+                logAsError: logLevel >= Options.LogToStandardErrorThreshold
+            );
+        }
+
+        private LogMessageEntry CreateLogMessageSystemd(StringBuilder logBuilder, LogLevel logLevel, string logName, int eventId, string message, Exception exception)
+        {
+            // systemd reads messages from standard out line-by-line in a '<pri>message' format.
+            // newline characters are treated as message delimiters, so we must replace them.
+            // Messages longer than the journal LineMax setting (default: 48KB) are cropped.
+            // Example:
+            // <6>ConsoleApp.Program[10] Request received
+
+            // loglevel
+            var logLevelString = GetSyslogSeverityString(logLevel);
+            logBuilder.Append(logLevelString);
+
+            // timestamp
+            var timestampFormat = Options.TimestampFormat;
+            if (timestampFormat != null)
+            {
+                logBuilder.Append(DateTime.Now.ToString(timestampFormat));
+            }
+
+            // category and event id
+            logBuilder.Append(logName);
+            logBuilder.Append("[");
+            logBuilder.Append(eventId);
+            logBuilder.Append("]");
+
+            // scope information
+            GetScopeInformation(logBuilder, multiLine: false);
+
+            // message
+            if (!string.IsNullOrEmpty(message))
+            {
+                logBuilder.Append(' ');
+                // message
+                var len = logBuilder.Length;
+                AppendAndReplaceNewLine(logBuilder, message);
+            }
+
+            // exception
+            // System.InvalidOperationException at Namespace.Class.Function() in File:line X
+            if (exception != null)
+            {
+                logBuilder.Append(' ');
+                AppendAndReplaceNewLine(logBuilder, exception.ToString());
+            }
+
+            // newline delimiter
+            logBuilder.Append(Environment.NewLine);
+
+            return new LogMessageEntry(
                 message: logBuilder.ToString(),
                 logAsError: logLevel >= Options.LogToStandardErrorThreshold
-            ));
+            );
 
-            logBuilder.Clear();
-            if (logBuilder.Capacity > 1024)
+            void AppendAndReplaceNewLine(StringBuilder sb, string message)
```

Comment:
```suggestion
            static void AppendAndReplaceNewLine(StringBuilder sb, string message)
```
Just to be on the safe side for not capturing any locals from the parent (i.e. no DisplayClass (which will be a struct :wink:)).